core/bootstrap/bootstrap.go

```diff
@@ -72,19 +72,40 @@ var DefaultBootstrapConfig = BootstrapConfig{
 	MaxBackupBootstrapSize:  20,
 }
 
-func BootstrapConfigWithPeers(pis []peer.AddrInfo) BootstrapConfig {
+// BootstrapConfigWithPeers creates a default BootstrapConfig configured with
+// the specified peers, and optional functions to load and save backup peers.
+func BootstrapConfigWithPeers(pis []peer.AddrInfo, options ...func(*BootstrapConfig)) BootstrapConfig {
 	cfg := DefaultBootstrapConfig
 	cfg.BootstrapPeers = func() []peer.AddrInfo {
 		return pis
 	}
+	for _, opt := range options {
+		opt(&cfg)
+	}
 	return cfg
 }
 
+// WithBackupPeers configures functions to load and save backup bootstrap peers.
+func WithBackupPeers(load func(context.Context) []peer.AddrInfo, save func(context.Context, []peer.AddrInfo)) func(*BootstrapConfig) {
+	if save == nil && load != nil || save != nil && load == nil {
+		panic("both load and save backup bootstrap peers functions must be defined")
+	}
+	return func(cfg *BootstrapConfig) {
+		cfg.LoadBackupBootstrapPeers = load
+		cfg.SaveBackupBootstrapPeers = save
```

Comment:
Oh and I realized, theses needs to be private now that you have an option.

Reply:
The problem with making these private is that they are accessed directly here, and possibly elsewhere:
https://github.com/ipfs/kubo/blob/master/core/core.go#L172-L173
https://github.com/ipfs/kubo/blob/master/core/core.go#L181-L182

Also, since those items are exported in the current release, making them private could be considered a breaking API change as it would break any other software that sets these, expecting them to be exported (public).

Reply:
We can fix the cases in Kubo.

> Also, since those items are exported in the current release, making them private could be considered a breaking API change as it would break any other software that sets these, expecting them to be exported (public).

It was already a breaking API change to add them and make it panic if nil.
We are not carefull about not breaking consumers inside Kubo, boxo is meant for the pieces of code that we are carefull to not break so consumers can import them.